crates/holochain/src/core/ribosome/host_fn/call.rs

```diff
@@ -275,4 +292,124 @@ pub mod wasm_test {
         install_app("bob_app", cell_data, vec![dna_file], handle.clone()).await;
         bob_cell_id
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    /// we can call a fn on a remote
+    async fn call_remote_test() {
+        // ////////////
+        // START DNA
+        // ////////////
+
+        let dna_def = DnaDef {
+            name: "call_remote_test".to_string(),
+            uid: "ba1d046d-ce29-4778-914b-47e6010d2faf".to_string(),
+            properties: SerializedBytes::try_from(()).unwrap(),
+            zomes: vec![TestWasm::WhoAmI.into()].into(),
+        };
+        let dna_file = DnaFile::new(dna_def, vec![TestWasm::WhoAmI.into()])
+            .await
+            .unwrap();
+
+        // //////////
+        // END DNA
+        // //////////
+
+        // ///////////
+        // START ALICE
+        // ///////////
+
+        let alice_agent_id = fake_agent_pubkey_1();
+        let alice_cell_id = CellId::new(dna_file.dna_hash().to_owned(), alice_agent_id.clone());
+        let alice_installed_cell = InstalledCell::new(alice_cell_id.clone(), "alice_handle".into());
+
+        // /////////
+        // END ALICE
+        // /////////
+
+        // /////////
+        // START BOB
+        // /////////
+
+        let bob_agent_id = fake_agent_pubkey_2();
+        let bob_cell_id = CellId::new(dna_file.dna_hash().to_owned(), bob_agent_id.clone());
+        let bob_installed_cell = InstalledCell::new(bob_cell_id.clone(), "bob_handle".into());
```

Comment:
This is very outdated way to write tests. Might be worth checking out some of the sweetest tests where you don't need to write any of this bioler plate.

Reply:
@freesig sure i just copied and pasted some old tests from a file i deleted, i could put the time into porting them while i'm here :)

Reply:
> This now allows calling remotely across dnas which wasn't possible in the previous call_remote. Is that what we actually want?

As originally planned, I thought the enum of call structures does not allow specifying a different dna target for a remote call, but rather was constrained to other agents on the same DNA.

Call should be able to specify a different target cell (dna+agent) when calling locally (formerly known as bridge type call),  But it looks to me on line 42 of call.rs, a remote call over the network is only able to pass in the agent id, not a different dna hash. As near as I can tell, this is per spec -- you can't call a different dna/dht via network/remote, only locally within the conductor.